File: in_Java/0389 Find the Difference.java
class Solution {
    public char findTheDifference(String s, String t) {
        int[] d = new int[26];
        int tt = 0;
        for(char c : s.toCharArray()){
            d[c - 'a']++;
        }
        for(char c : t.toCharArray()){
            d[c - 'a']--;
        }
        for(int i = 0; i < 26; i++){
            if(d[i] != 0){
                tt = i;
                break;
            }
        }
        return (char)((int)'a' + tt);
    }
}

File: in_Python/0389 Find the Difference.py
class Solution(object):
    def findTheDifference(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: str
        """
        d = {}
        for si in s:
            if si not in d:
                d[si] = 1
            else:
                d[si] += 1
        res = ''
        for ti in t:
            if ti not in d:
                return ti
            else:
                d[ti] -= 1
                if d[ti] < 0:
                    res = ti
                    break
        return res